js/src/sdk/base.toolset.ts

```diff
@@ -308,12 +308,20 @@ export class ComposioToolSet {
         connectionId: accountId,
       });
     }
-
+    if (Object.keys(this.connectedAccountIds).length > 0) {
+      const actionDetails = await this.actions.get({ actionName: action });
+      const appNameKey = Object.keys(this.connectedAccountIds).find(
```

Comment:
You are calling `Object.keys(this.connectedIds)` twice, ideally store it in a variable instead and reuse.

Reply:
Fixed it!